testing/jormungandr-scenario-tests/src/interactive/args/mod.rs

```diff
@@ -105,23 +123,42 @@ impl UserInteractionController {
             .find(|x| x.address() == address)
             .unwrap_or_else(|| panic!("cannot find wallet with alias: {}", wallet_alias));
 
-        let check = self.controller.fragment_sender().send_vote_cast(
-            wallet,
-            &vote_plan_def.into(),
-            proposal_index as u8,
-            &Choice::new(choice),
-            via,
-        )?;
+        let node = self.nodes.iter().find(|x| x.alias() == node_alias);
+        let legacy_node = self.legacy_nodes.iter().find(|x| x.alias() == node_alias);
+
+        let check = match (node, legacy_node) {
+            (Some(node), None) => self.controller.fragment_sender().send_vote_cast(
+                wallet,
+                &vote_plan_def.into(),
+                proposal_index as u8,
+                &Choice::new(choice),
+                node,
+            )?,
+            (None, Some(node)) => self.controller.fragment_sender().send_vote_cast(
+                wallet,
+                &vote_plan_def.into(),
+                proposal_index as u8,
+                &Choice::new(choice),
+                node,
+            )?,
```

Comment:
The arm bodies are the same. Use

```suggestion
            (Some(node), None) | (None, Some(node)) => self.controller.fragment_sender().send_vote_cast(
                wallet,
                &vote_plan_def.into(),
                proposal_index as u8,
                &Choice::new(choice),
                node,
            )?,
```

Reply:
Those bindings are of different types, I don't like this solution but I do not know how that could be rewritten in a better fashion

Reply:
Ah, I did not realize there's static polymorphism over the parameter.